### PR TITLE
feat(REST): storev3 client

### PR DESCRIPTION
**Pull request by richard-ramos**
# Description
Adds an endpoint for storev3

### CODE REVIEW

File: cmd/waku/server/rest/waku_rest.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func NewWakuRest(node *node.WakuNode, config RestConfig, log *zap.Logger) *WakuR
 	_ = NewDebugService(node, mux)
 	_ = NewHealthService(node, mux)
 	_ = NewStoreService(node, mux)
```
Comment (chaitanyaprem) [inline cmd/waku/server/rest/waku_rest.go:54]: Maybe rename to StoreQuery?

File: cmd/waku/server/rest/store.go [hunk 2]
```diff
@@ -66,72 +48,78 @@ func getStoreParams(r *http.Request) (*legacy_store.Query, []legacy_store.Histor
...
+				return nil, nil, err
+			}
+			hashes = append(hashes, pb.ToMessageHash(hash))
+		}
+	}
+
+	isMsgHashCriteria := false
+	if len(hashes) != 0 {
+		isMsgHashCriteria = true
+		if pubsubTopic != "" {
```
Comment (chaitanyaprem) [inline cmd/waku/server/rest/store.go:88]: Shouldn't we include contentTopic check as well?

**Review by chaitanyaprem — APPROVED** [review]
Approving it as only minor comments required to be addressed.

File: cmd/waku/server/rest/store.go [hunk 3]
```diff
@@ -12,52 +12,34 @@ import (
...
-type StoreWakuMessage struct {
-	Payload      []byte  `json:"payload"`
-	ContentTopic string  `json:"contentTopic"`
-	Version      *uint32 `json:"version,omitempty"`
-	Timestamp    *int64  `json:"timestamp,omitempty"`
-	Meta         []byte  `json:"meta,omitempty"`
-}
-
-const routeStoreMessagesV1 = "/store/v1/messages"
+const routeStoreMessagesV1 = "/store/v3/messages"
```
[the remark this answers is not in the document]
Comment (richard-ramos) [reply cmd/waku/server/rest/store.go:28]: Probably a good idea to change that in nwaku as well:
https://github.com/waku-org/nwaku/blob/master/waku/waku_api/rest/store/handlers.nim#L184

**Pull request merged**